core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2482,19 +2490,25 @@ public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFile
     if ((ownerGroupChanged || permissionChanged) && inode.isPersisted()) {
       MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
       String ufsUri = resolution.getUri().toString();
+      if (CommonUtils.isUfsObjectStorage(ufsUri)) {
+        throw new AccessControlException(
+            "setOwner/setMode is not supported to object storage UFS via Alluxio. UFS: " + ufsUri);
+      }
```

Comment:
Is `AccessControlException` the best type to throw here? Perhaps `NotSupportedException`?